src/FNA3D_Driver_Vulkan.c

```diff
@@ -6499,29 +6519,50 @@ static void VULKAN_SetBlendState(
 ) {
 	VulkanRenderer *renderer = (VulkanRenderer*) driverData;
 
-	SDL_memcpy(&renderer->blendState, blendState, sizeof(FNA3D_BlendState));
-
-	/* Dynamic state */
 	VULKAN_SetBlendFactor(
 		driverData,
 		&blendState->blendFactor
 	);
+
 	VULKAN_SetMultiSampleMask(
 		driverData,
 		blendState->multiSampleMask
 	);
+
+	SDL_memcpy(&renderer->blendState, blendState, sizeof(FNA3D_BlendState));
 }
 
 static void VULKAN_SetDepthStencilState(
 	FNA3D_Renderer *driverData,
 	FNA3D_DepthStencilState *depthStencilState
 ) {
 	VulkanRenderer *renderer = (VulkanRenderer*) driverData;
-	SDL_memcpy(
-		&renderer->depthStencilState,
-		depthStencilState,
-		sizeof(FNA3D_DepthStencilState)
-	);
+
+	if (renderer->depthStencilState.stencilMask != depthStencilState->stencilMask ||
```

Comment:
Since we're storing with memcpy we can probably replace all this with memcmp to keep it short.

Reply:
Oh, i didn't know about memcmp. Neat! I can rewrite some of these to use that func instead